crates/metaslang/cst/src/query/engine.rs

```diff
@@ -29,37 +31,44 @@ impl<T: KindTypes + 'static> Cursor<T> {
         match self.node() {
             Node::<T>::Rule(rule) => match node_selector {
                 NodeSelector::Anonymous => true,
-                NodeSelector::Kind { kind } => Kind::Rule(rule.kind) == *kind,
-                NodeSelector::Text { .. } => false,
-                NodeSelector::Label { label } => Some(*label) == self.label(),
-                NodeSelector::LabelAndKind { label, kind } => {
-                    Some(*label) == self.label() && Kind::Rule(rule.kind) == *kind
+                NodeSelector::NodeKind { node_kind: kind } => {
```

Comment:
I think a few instances like these missed the rename:

```suggestion
                NodeSelector::NodeKind { node_kind } => {
```

Reply:
I fixed these in the rebase